packages/lesswrong/server/manualMigrations/2025-02-26-fieldChangesPopulateTable.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable no-console */
+import { registerMigration } from "./migrationUtils";
+import LWEvents from "@/lib/collections/lwevents/collection";
+import { executePromiseQueue, sleep } from "@/lib/utils/asyncUtils";
+import { randomId } from "@/lib/random";
+import { FieldChanges } from "@/lib/collections/fieldChanges/collection";
+
+export default registerMigration({
+  name: "fieldChangesPopulateTable",
+  dateWritten: "2025-02-26",
+  idempotent: true,
+  action: async () => {
+    // Transfers field-changes from LWEvents to the dedicated FieldChanges
+    // collection. Note that LWEvents is an _extremely_ large table, mostly due
+    // to event types that are not this one, and queries against it must be
+    // made with great care lest they time out and cause problems for the
+    // database server.
+    // Also note that development DBs may be created by cloning a production
+    // DB with LWEvents omitted, which complicates performance testing. 
+
+    console.log("Loading field changes from the LWEvents table");
+    const fieldChangesFromLWEvents = await LWEvents.find({
+      name: "fieldChanges"
+    }).fetch();
+    
+    console.log(`Got ${fieldChangesFromLWEvents.length} field-change events`);
+    console.log("Inserting field changes into the FieldChanges table");
+    await executePromiseQueue(fieldChangesFromLWEvents.map((lwEvent) => async () => {
+      const changeGroupId = randomId();
+      for (const fieldName of Object.keys(lwEvent.properties.after)) {
+        await FieldChanges.rawInsert({
+          userId: lwEvent.userId,
+          changeGroup: changeGroupId,
+          documentId: lwEvent.documentId,
+          fieldName,
+          oldValue: lwEvent.properties.before[fieldName],
+          newValue: lwEvent.properties.after[fieldName],
```

Comment:
Possibly these need to JSON.stringify the before/after values if we're going with string-typed fields, but need to verify (and might still be worth trying to convert it to JSON)

Reply:
Nvm, fixed by fixing the insert & update query builders to correctly convert primitives to JSON values with the right type hints and changing the schema field types to `Object`.